pkg/connector/handletwit.go

```diff
@@ -0,0 +1,179 @@
+					Sender:      networkid.UserID(evtData.Executor.IDStr),
+				},
+				LogContext: func(c zerolog.Context) zerolog.Context {
+					return c.
+						Str("conversation_id", evtData.Conversation.ConversationID).
+						Str("new_name", evtData.Name).
+						Str("changed_by_user_id", evtData.Executor.IDStr)
+				},
+				PortalKey: tc.MakePortalKey(evtData.Conversation),
+				Timestamp: evtData.UpdatedAt,
+			},
+			ChatInfoChange: &bridgev2.ChatInfoChange{
+				ChatInfo: &bridgev2.ChatInfo{
+					Name: &evtData.Name,
+				},
+			},
+		}
+		tc.connector.br.QueueRemoteEvent(tc.userLogin, portalUpdateRemoteEvent)
+	case event.XEventParticipantsJoined:
+		portalMembersAddedRemoteEvent := &simplevent.ChatInfoChange{
+			EventMeta: simplevent.EventMeta{
+				Type: bridgev2.RemoteEventChatInfoChange,
+				LogContext: func(c zerolog.Context) zerolog.Context {
+					return c.
+						Str("conversation_id", evtData.Conversation.ConversationID).
+						Int("total_new_members", len(evtData.NewParticipants))
+				},
+				PortalKey: tc.MakePortalKey(evtData.Conversation),
+				Timestamp: evtData.EventTime,
+			},
+			ChatInfoChange: &bridgev2.ChatInfoChange{
+				MemberChanges: tc.UsersToMemberList(evtData.NewParticipants),
+			},
+		}
+		tc.connector.br.QueueRemoteEvent(tc.userLogin, portalMembersAddedRemoteEvent)
+	case event.XEventConversationDelete:
+		portalDeleteRemoteEvent := &simplevent.ChatDelete{
+			EventMeta: simplevent.EventMeta{
+				Type:      bridgev2.RemoteEventChatDelete,
+				PortalKey: tc.MakePortalKeyFromID(evtData.ConversationID),
+				LogContext: func(c zerolog.Context) zerolog.Context {
+					return c.
+						Str("conversation_id", evtData.ConversationID)
+				},
+				Timestamp: evtData.DeletedAt,
+			},
+			OnlyForMe: true,
+		}
+		tc.connector.br.QueueRemoteEvent(tc.userLogin, portalDeleteRemoteEvent)
+		tc.client.Logger.Info().Any("data", evtData).Msg("Deleted conversation")
+	default:
+		tc.client.Logger.Warn().Any("event_data", evtData).Msg("Received unhandled event case from twitter library")
+	}
+}
+
+func (tc *TwitterClient) wrapReaction(data event.XEventReaction) *simplevent.Reaction {
+	var eventType bridgev2.RemoteEventType
+	if data.Action == types.MessageReactionRemove {
+		eventType = bridgev2.RemoteEventReactionRemove
+	} else {
+		eventType = bridgev2.RemoteEventReaction
+	}
+
+	var receiver networkid.UserLoginID
+	if data.Conversation.Type == types.ONE_TO_ONE {
+		receiver = networkid.UserLoginID(tc.userLogin.ID)
+	}
+	return &simplevent.Reaction{
+		EventMeta: simplevent.EventMeta{
+			Type: eventType,
+			LogContext: func(c zerolog.Context) zerolog.Context {
+				return c.
+					Str("message_id", data.MessageID).
+					Str("sender", data.SenderID).
+					Str("reaction_key", data.ReactionKey).
+					Str("emoji_reaction", data.EmojiReaction)
+			},
+			PortalKey: networkid.PortalKey{
+				ID:       networkid.PortalID(data.Conversation.ConversationID),
+				Receiver: receiver,
+			},
```

Comment:
Portal keys should not be constructed manually, always use the appropriate helper method